recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -208,7 +220,7 @@ def generate(self):
         tc.cache_variables["BUILD_BENCHMARK"] = False
         tc.cache_variables["WITH_EXAMPLES"] = False
         tc.cache_variables["WITH_NO_DEPRECATED_CODE"] = self.options.with_no_deprecated_code
-        tc.cache_variables["WITH_STL"] = self.options.with_stl
+        tc.cache_variables["WITH_STL"] = self._default_stl_version
```

Comment:
I think we need a `self.cpp_info.components[…].defines.append(f”OPENTELEMETRY_STL_VERSION=20{self._default_cppstd}”)` in `package_info()`

Reply:
Thanks I've added this line. Should we keep `self.cpp_info.components["opentelemetry_common"].defines.append("HAVE_CPP_STDLIB")` as it is or add a version check?